src/main/java/appeng/recipes/factories/recipes/DamagedRecipeFactory.java

```diff
@@ -0,0 +1,88 @@
+
+package appeng.recipes.factories.recipes;
+
+
+import java.util.Iterator;
+
+import javax.annotation.Nonnull;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParseException;
+
+import net.minecraft.inventory.InventoryCrafting;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.IRecipe;
+import net.minecraft.item.crafting.Ingredient;
+import net.minecraft.util.JsonUtils;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.common.crafting.CraftingHelper;
+import net.minecraftforge.common.crafting.IRecipeFactory;
+import net.minecraftforge.common.crafting.JsonContext;
+import net.minecraftforge.oredict.ShapelessOreRecipe;
+
+
+public class DamagedRecipeFactory implements IRecipeFactory
+{
+	@Override
+	public IRecipe parse( JsonContext context, JsonObject json )
+	{
+		String group = JsonUtils.getString( json, "group", "" );
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "group" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)